### PR TITLE
update hovered source on all crosshair changes

### LINKED ISSUE #1523: `hoveredObjectId` (`subscribeCrosshairMove`) is not updated when chart is scrolled horizontally
**Lightweight Charts™ Version:** 4.2.0

**Steps/code to reproduce:**

1. Open demo https://jsfiddle.net/p5cbjtu0/5/
2. Open console
3. Hover over any marker and check `hoveredObjectId` in the console (there should be ID)
4. Scroll the chart horizontally and notice `hoveredObjectId` does not change unlike `logical`

**Actual behavior:**

`hoveredObjectId` does not change when chart is scrolled.

**Expected behavior:**

`hoveredObjectId` should change and reflect the currently hovered marker (object). If there is no hovered marker, `hoveredObjectId` should be `undefined`.

**Screenshots:**

Handler:
image

Hovered marker (orange):
image

Scrolled chart (`logical` has changed while `hoveredObjectId` hasn't):
image

**CodeSandbox/JSFiddle/etc link:** https://jsfiddle.net/p5cbjtu0/5/

### PR DESCRIPTION
**Pull request by illetid**
**Type of PR:**  enhancement

**PR checklist:**

- [x] Addresses an existing issue: fixes #1523
- [x] Includes tests
- [ ] Documentation update

**Overview of change:**
- moved the hitTest check into the model to update the hovered source on `scrollToPosition` and other chart changes
- added hitTest for the arrow markers

### CODE REVIEW

**Comment by SlicedSilver** [comment]
Could you try add an interaction e2e test case which would fail on the old code, and pass on this code?
and could you check if this PR addresses any of the other `hoveredObjectId` bugs in the v5 milestone? 🙏

**Comment by illetid** [comment]
> Could you try add an interaction e2e test case which would fail on the old code, and pass on this code? and could you check if this PR addresses any of the other `hoveredObjectId` bugs in the v5 milestone? 🙏

@SlicedSilver other two bugs #1503 and  #1513 were caused by overlapping with price line, and were resolved when we moved series markers into the plugin. I'll add tests for all 3 cases

File: tests/e2e/interactions/test-cases/markers/hit-test-priceline-overlap.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+function afterInitialInteractions() {
+	return new Promise(resolve => {
+		requestAnimationFrame(resolve);
+	});
+}
+
+function afterFinalInteractions() {
+	const pass = lastHoveredObjectId === 'TEST';
+	// throw new Error(`Expected hoveredObjectId to be equal to 'TEST'. Actual: ${markerX} = ${markerY}`);
```
Comment (SlicedSilver) [inline tests/e2e/interactions/test-cases/markers/hit-test-priceline-overlap.js:93]: Lets remove this comment

Comment (illetid) [reply tests/e2e/interactions/test-cases/markers/hit-test-priceline-overlap.js]: removed, thanks

**Review by SlicedSilver — APPROVED** [review]

**Pull request merged**